sphinx_gallery/gen_rst.py

```diff
@@ -565,6 +565,13 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf):
     time_elapsed = 0
     block_vars = {'execute_script': execute_script, 'fig_count': 0,
                   'image_path': image_path_template, 'src_file': src_file}
+
+    argv_orig = sys.argv[:]
+    if block_vars['execute_script']:
+        print('Executing file %s' % src_file)
+        sys.argv[0] = src_file
```

Comment:
Is this necessary? `sys.argv[0]` corresponds to the executing binary, most probably it would be `python` as example scripts are not executable on their own (there is no shebang). I wouldn't like to have this hint to the `src_file`, because people might start abusing it. We already don't provide the `__file__` variable because of incompatibility with the Jupyter notebooks.